betting-game-react/src/components/Navbar/Navbar.test.tsx

```diff
@@ -0,0 +1,148 @@
+import '@testing-library/jest-dom/extend-expect';
+
+import { createMemoryHistory } from 'history';
+import React from 'react';
+import { MemoryRouter, Router } from 'react-router-dom';
+
+import { fireEvent, render } from '@testing-library/react';
+
+import { ABOUTUS, LOGIN, MATCHES, STANDINGS } from '../../config/routes';
+import Navbar from './Navbar';
+import { NavbarButtonProps, navbarButtons } from './NavbarButtonProps';
+
+describe('<Navbar />', () => {
+  test('it should render a navbar with a logo, a login button and all navigation buttons', () => {
+    const history = createMemoryHistory();
+    history.push('/bad/route');
+    const { getByTestId } = render(
+      <MemoryRouter>
+        <Navbar />
+      </MemoryRouter>
+    );
+
+    const navbar = getByTestId('Navbar');
+    const navbarLogo = getByTestId('NavbarLogo');
+    let navbarButtonsArray: any[] = [];
+    navbarButtons.map((navbarButton: NavbarButtonProps, index: number) => {
+      navbarButtonsArray.push(
+        getByTestId(`${navbarButton.buttonTestIdPrefix}-NavButton`)
+      );
+    });
+
+    expect(navbar).toBeInTheDocument();
+    expect(navbarLogo).toBeInTheDocument();
+    navbarButtonsArray.forEach((navbarButton) => {
+      expect(navbarButton).toBeInTheDocument();
+    });
```

Comment:
Hmmmm. Yes, this is probably fine.
Personally I prefer my expects to be as explicit as possible for example :
`expect(firstButton).toBeInTheDocument()...` etc. as you'd immediately see which button is going to fail the test etc, it also allows you to split up the test easier if that need arises.

It's a bit about personal preference here 🤷